### PR TITLE
chore: add kotlin update to docs and formatting changes for flutter

### PR DESCRIPTION
**Pull request by khatruong2009**
#### Description of changes:
Update Flutter docs:
- change kotlin version after creating a Flutter app
- adjust for formatting changes for Flutter's new Gradle plugins

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [x] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [x] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [x] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/project-setup/flutter/platform-setup/android.mdx [hunk 1]
```diff
@@ -10,10 +12,23 @@ From your project root, navigate to the `android/` directory and open `build.gra
...
 
-If your Kotlin Gradle plugin version is below 1.7, update `android/build.gradle`:
+If your Kotlin Gradle plugin version is below 1.9, update `android/build.gradle`:
 
 ```
-ext.kotlin_version = '1.7.10'
+ext.kotlin_version = '1.9.10'
+```
+
+If your Flutter app was generated with Flutter 3.16 or higher, make the following changes instead:
```
[the remark this answers is not in the document]
Comment (Jordan-Nelson) [reply src/fragments/lib/project-setup/flutter/platform-setup/android.mdx:21]: Agreed

[the remark this answers is not in the document]
Comment (khatruong2009) [reply src/fragments/lib/project-setup/flutter/platform-setup/android.mdx:21]: In the flutter apps created by an older version of flutter, the kotlin version is stored in `build.gradle` but after 3.16, it got moved to `settings.gradle`

[the remark this answers is not in the document]
Comment (Jordan-Nelson) [reply src/fragments/lib/project-setup/flutter/platform-setup/android.mdx:21]: I think w should we just tell customers to follow [this migration guide](https://docs.flutter.dev/release/breaking-changes/flutter-gradle-plugin-apply) if they created their project with Flutter 3.16 or lower, and then only provide the info for the latest version.

File: src/fragments/lib/project-setup/flutter/platform-setup/android.mdx [hunk 2]
```diff
@@ -1,19 +1,27 @@
-Amplify requires a minimum of API level 24 (Android 7.0), Gradle 7 and Kotlin > 1.7 when targeting Android.
+Amplify requires a minimum of API level 24 (Android 7.0), Gradle 7 and Kotlin > 1.9 when targeting Android.
 
-From your project root, navigate to the `android/` directory and open `build.gradle` in the text editor of your choice. Update the Gradle plugin version to 7.4.2 or higher:
+_If your Flutter app was generated with Flutter 3.16 or lower, please follow the migration guide [here](https://docs.flutter.dev/release/breaking-changes/flutter-gradle-plugin-apply)._
+
+From your project root, navigate to the `android/` directory and open `settings.gradle` in the text editor of your choice. Update the Gradle plugin version to 7.4.2 or higher:
```
Comment (Equartey) [inline src/fragments/lib/project-setup/flutter/platform-setup/android.mdx:5]:
```suggestion
From your project root, navigate to the `android/` directory and open `build.gradle` in the text editor of your choice. Update the Gradle plugin version to 7.4.2 or higher:
```

Comment (Equartey) [reply src/fragments/lib/project-setup/flutter/platform-setup/android.mdx:5]: Discussed offline, correct location is `build.gradle`

**Review by Jordan-Nelson — APPROVED** [review]

**Review by Equartey — APPROVED** [review]

**Review by salihgueler — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Pull request merged**